### DISCUSSION TITLE
Two contourlines in "Eddy detection : Antartic Circumpolar Current"

### DISCUSSION THREAD

**Discussion by lydi-keppler** [AntSimi/py-eddy-tracker#188] [General]
Hi,

I'm looking at the script "Eddy detection: Antarctic Circumpolar Current" from [this link](https://py-eddy-tracker.readthedocs.io/en/latest/python_module/02_eddy_identification/pet_eddy_detection_ACC.html). In the third figure (comparison between two detections), some eddies have two contour lines (an inner solid one and an outer dashed one). This is the case in both plots on the left and right (i.e., with and without the filter), so the two contour lines probably have nothing to do with the filter. My guess is one is the outermost contour line from the ADT, and one is the speed radius, is this correct? And how do I only plot one of them (the ADT-based eddy edge)?  Sorry, I'm relatively new to Python ...

**Comment by AntSimi** [comment]
Outermost contour could be also call effective contour, it's the last closed contour around the eddy. Inner contour call also speed contour, along this contour mean speed is at the highest value. Sometime effective and speed contour could be the same.
Display options could be found here : https://py-eddy-tracker.readthedocs.io/en/latest/_autosummary/py_eddy_tracker.observations.observation.EddiesObservations.html#py_eddy_tracker.observations.observation.EddiesObservations.display

**Reply by lydi-keppler** [reply]
Fantastic, thank you!

**Comment by lydi-keppler** [comment]
Hi again,

I'm doing some work with the Eddy Py-Tracker and love it! However, I have a question I hope you can help me with: How do I filter out small features, e.g., eddies with a radius smaller than 50 km?

I have tried the following after running the identification <a, c = g.eddy_identification("adt", **kw_ident)>:
a = np.where(a.radius_e > 50000)
but it doesn't work as it destroys the structure of a. Any thoughts? Thanks! 

Lydi

**Comment by AntSimi** [comment]
Hi,

Following method could be use
[https://py-eddy-tracker.readthedocs.io/en/latest/_autosummary/py_eddy_tracker.observations.observation.EddiesObservations.html#py_eddy_tracker.observations.observation.EddiesObservations.extract_with_mask](https://www.researchgate.net/deref/https%3A%2F%2Fpy-eddy-tracker.readthedocs.io%2Fen%2Flatest%2F_autosummary%2Fpy_eddy_tracker.observations.observation.EddiesObservations.html%23py_eddy_tracker.observations.observation.EddiesObservations.extract_with_mask?_tp=eyJjb250ZXh0Ijp7ImZpcnN0UGFnZSI6Il9kaXJlY3QiLCJwYWdlIjoiX2RpcmVjdCJ9fQ)

Antoine

**Comment by lydi-keppler** [comment]
Fantastic, that worked, thank you! I really appreciate your help.

Merci!
Lydi

On Fri, 15 Dec 2023 at 10:12, Antoine Delepoulle ***@***.***>
wrote:

> Hi,
>
> Following method could be use
>
> https://py-eddy-tracker.readthedocs.io/en/latest/_autosummary/py_eddy_tracker.observations.observation.EddiesObservations.html#py_eddy_tracker.observations.observation.EddiesObservations.extract_with_mask
> <https://www.researchgate.net/deref/https%3A%2F%2Fpy-eddy-tracker.readthedocs.io%2Fen%2Flatest%2F_autosummary%2Fpy_eddy_tracker.observations.observation.EddiesObservations.html%23py_eddy_tracker.observations.observation.EddiesObservations.extract_with_mask?_tp=eyJjb250ZXh0Ijp7ImZpcnN0UGFnZSI6Il9kaXJlY3QiLCJwYWdlIjoiX2RpcmVjdCJ9fQ>
>
> Antoine
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AntSimi/py-eddy-tracker/discussions/188#discussioncomment-7866845>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASLO4XQRD2EMIOS6KALGZ2DYJSHINAVCNFSM6AAAAAAUXBXD5KVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TQNRWHA2DK>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
><((((º>
